scalalib/src/mill/scalalib/CoursierModule.scala

```diff
@@ -33,6 +33,34 @@ trait CoursierModule extends mill.Module {
     Lib.depToDependencyJava(_: Dep)
   }
 
+  /**
+   * A `CoursierModule.Resolver` to resolve dependencies.
+   *
+   * Unlike `defaultResolver`, this resolver can resolve Mill internal modules
+   * (obtained via `JavaModule#coursierDependency`).
+   *
+   * @return `CoursierModule.Resolver` instance
+   */
+  def internalResolver: Task[CoursierModule.Resolver] = Task.Anon {
```

Comment:
How about `coursierModuleResolver`. It's longish, but better tells what it is. Also, it's prefixed with the originating trait name, which make it easy to associate with.

Reply:
Or `internalModuleResolver`, if you want to keep it close to `internalRepositories`.

Reply:
I'm not sure about `coursierModuleResolver`, as in Mill, "module" is associated with Mill modules, while the resolver resolves mainly external dependencies. `coursierDependencyResolver` or just `dependencyResolver` / `depResolver` might work too.

Reply:
Renamed it to `internalCoursierResolver` here: https://github.com/com-lihaoyi/mill/pull/4472

Reply:
I'll report it here once the state of https://github.com/com-lihaoyi/mill/pull/4472 is ok